tests/js/session-tests.js

```diff
@@ -1033,42 +1034,66 @@ module.exports = {
         });
     },
 
-    testClientResetAtOpen() {
+    async testClientResetAtOpen() {
         // FIXME: try to enable for React Native
         if (!isNodeProcess) {
             return;
         }
-
-        return Realm.Sync.User.login('http://127.0.0.1:9080', Realm.Sync.Credentials.anonymous()).then(user => {
-            return new Promise((resolve, _reject) => {
-                var realm;
-                const config = user.createConfiguration({ sync: { url: 'realm://127.0.0.1:9080/~/myrealm' } });
-                config.sync.clientResyncMode = 'manual';
-                config.sync.error = (sender, error) => {
-                    try {
-                        TestCase.assertEqual(error.name, 'ClientReset');
-                        TestCase.assertDefined(error.config);
-                        TestCase.assertNotEqual(error.config.path, '');
-                        const path = realm.path;
-                        realm.close();
-                        Realm.Sync.initiateClientReset(path);
-                        // open Realm with error.config, and copy required objects a Realm at `path`
-                        resolve();
-                    }
-                    catch (e) {
-                        _reject(e);
-                    }
-                };
-
-                return Realm.open(config).then(r => {
-                    realm = r;
-                    const session = realm.syncSession;
-
-                    TestCase.assertEqual(session.config.error, config.sync.error);
-                    session._simulateError(211, 'ClientReset'); // 211 -> divering histories
-                });
-            });
+        const fetch = require('node-fetch');
+
+        let user = await Realm.Sync.User.login('http://127.0.0.1:9080', Realm.Sync.Credentials.nickname('admin', true));
```

Comment:
nickname admin login is deprecated. I think we should stop using it.

Reply:
lets ignore this

Reply:
True, we need to find a better way to generate a user.

Reply:
there is one already there. We are using it from here. https://github.com/realm/realm-js/blob/master/tests/js/admin-user-helper.js. 
But most probably you can ignore this suggestion